packages/foundry/contracts/witch/WitchV2.sol

```diff
@@ -375,6 +376,12 @@ contract WitchV2 is AccessControl {
             auction_ = _auction(vault, series, balances, debt);
         }
 
+        // 0 value is to offer a nice API for people that wants to pay all
+        // GT check is to cater for partial buys right before this method executes
+        artIn = (maxArtIn == 0 || maxArtIn > auction_.art)
```

Comment:
Why not remove the use of zero as a magic number and let the user enter MAX_UINT256?

Reply:
I think that 0 makes a "nicer" API, but I don't have any other strong justification

Reply:
I prefer to avoid magic values and edge cases wherever possible. I also think that zero as a magic value is quite dangerous, as you can easily obtain it by accident.

Reply:
I get that this is a view function, but still.

Reply:
Fair enough, will change it, I don't have any strong reasons other than cosmetics :)

Reply:
On my defense, the comment about 0 meaning all was there, I just made it work :P

Reply:
I'm happy to agree that I often take decisions that I later regret, so the original comment might have been mine.

Reply:
LOL, no worries, it's gone now